src/components/FormFields/DataIntegrityChecksField.js

```diff
@@ -0,0 +1,158 @@
+
+const { Field, useField } = ReactFinalForm
+
+const VALIDATOR = value =>
+    value && value.length < 1
+        ? i18n.t('Please select checks to run.')
+        : undefined
+
+const DataIntegrityChecksField = ({ label, name }) => {
+    const options = hooks.useParameterOptions('dataIntegrityChecks')
+    const {
+        input: { value, onChange },
+    } = useField(name)
+
+    const hasValue = !!value && value.length > 0
+    const [runSelected, setRunSelected] = useState(hasValue)
+
+    const translatedOptions = options
+        .map(option => ({
+            ...option,
+            value: option.name,
+            label: getCheckName(option.name),
+            severity: severityMap[option.severity],
+        }))
+        .sort((a, b) => a.label.localeCompare(b.label))
+
+    const toggle = ({ value }) => {
+        const checked = value === 'true'
+
+        if (!checked) {
+            // clear checks when "Run all" is selected
+            onChange([])
+        }
+        setRunSelected(checked)
+    }
+
+    return (
+        <FieldGroup label={i18n.t('Checks to run')}>
+            <Radio
+                name={'checksToRun'}
+                value={String(false)}
+                label={i18n.t('Run all available checks')}
+                checked={!runSelected}
+                onChange={toggle}
+            />
+            <Radio
+                name={'checksToRun'}
+                value={String(true)}
+                label={i18n.t('Only run selected checks')}
+                checked={runSelected}
+                onChange={toggle}
+            />
+            <Field
+                name={name}
+                component={ChecksTransfer}
+                options={translatedOptions}
+                label={label}
+                validate={VALIDATOR}
+                // conditional rendering of FinalForm-fields cause some issues,
+                // see https://github.com/final-form/react-final-form/issues/809
+                hidden={!runSelected}
+            />
+        </FieldGroup>
+    )
+}
+
+const LabelComponent = ({ label, severity, highlighted, disabled }) => (
+    <div
+        className={cx(styles.transferOption, {
+            [styles.highlighted]: highlighted,
+            [styles.disabled]: disabled,
+        })}
+    >
+        <div className={styles.optionName}>{label}</div>
+        <div
+            className={cx(styles.optionSeverity, {
+                [styles.highlighted]: highlighted,
+            })}
+        >{`${i18n.t('Severity')}: ${severity}`}</div>
+    </div>
+)
+
+LabelComponent.propTypes = TransferOption.propTypes
+
+const renderOption = option => (
+    <TransferOption {...option} label={<LabelComponent {...option} />} />
+)
+
+const ChecksTransfer = ({ input, meta, options = [], hidden }) => {
+    const { onChange } = input
+
+    const handleChange = useCallback(
+        ({ selected }) => {
+            onChange(selected)
+        },
+        [onChange]
+    )
```

Comment:
I'm curious, why not use an inline fn in the transfer's onchange directly?

Reply:
Think I did this because I was debugging some re-renders and weird behaviour for the transfer-component, and just left it as is. I mean, it's probably not necessary, but I'm also curious why you would prefer inline, as this just means potentially less re-renders?

Reply:
These articles have always been what I personally refer to with regards to inline functions react components: 

- https://medium.com/@ryanflorence/react-inline-functions-and-performance-bdff784f5578 
- https://kentcdodds.com/blog/usememo-and-usecallback.

It doesn't always make things better so I just refrain from using them for performance reasons until there's a clear need for them.